asf_public_discourse_home_decarbonisation/utils/plotting_utils.py

```diff
@@ -0,0 +1,466 @@
+    """
+    Adds titles, subtitles, configures font sizes andadjusts gridlines
+    """
+    return (
+        fig.properties(
+            title={
+                "anchor": "start",
+                "text": chart_title,
+                "fontSize": FONTSIZE_TITLE,
+                "subtitle": chart_subtitle,
+                "subtitleFont": FONT,
+                "subtitleFontSize": FONTSIZE_SUBTITLE,
+            },
+        )
+        .configure_axis(
+            gridDash=[1, 7],
+            gridColor="grey",
+            labelFontSize=FONTSIZE_NORMAL,
+            titleFontSize=FONTSIZE_NORMAL,
+        )
+        .configure_legend(
+            titleFontSize=FONTSIZE_NORMAL,
+            labelFontSize=FONTSIZE_NORMAL,
+        )
+        .configure_view(strokeWidth=0)
+    )
+
+
+def finding_path_to_font(font_name: str):
+    """
+    Finds path to specific font.
+    Args:
+        font_name: name of font
+    """
+
+    all_font_files = font_manager.findSystemFonts()
+    font_files = [f for f in all_font_files if font_name in f]
+    if len(font_files) == 0:
+        font_files = [f for f in all_font_files if "DejaVuSans.ttf" in f]
+    return font_files[0]
+
+
+def set_spines():
+    """
+    Function to add or remove spines from plots.
+    """
+    mpl.rcParams["axes.spines.left"] = True
+    mpl.rcParams["axes.spines.right"] = False
+    mpl.rcParams["axes.spines.top"] = False
+    mpl.rcParams["axes.spines.bottom"] = True
+
+
+def set_plotting_styles():
+    """
+    Function that sets plotting styles.
+    """
+
+    sns.set_context("talk")
+
+    set_spines()
+
+    # Had trouble making it find the font I set so this was the only way to do it
+    # without specifying the local filepath
+    all_font_files = font_manager.findSystemFonts()
+
+    try:
+        mpl.rcParams["font.family"] = "sans-serif"
+        font_files = [f for f in all_font_files if "Averta-Regular" in f]
+        for font_file in font_files:
+            font_manager.fontManager.addfont(font_file)
+        mpl.rcParams["font.sans-serif"] = "Averta"
+    except:
+        print("Averta" + " font could not be located. Using 'DejaVu Sans' instead")
+        font_files = [f for f in all_font_files if "DejaVuSans.ttf" in f][0]
+        for font_file in font_files:
+            font_manager.fontManager.addfont(font_file)
+        mpl.rcParams["font.family"] = "sans-serif"
+        mpl.rcParams["font.sans-serif"] = "DejaVu Sans"
+
+    mpl.rcParams["xtick.labelsize"] = 18
+    mpl.rcParams["ytick.labelsize"] = 18
+    mpl.rcParams["axes.titlesize"] = 20
+    mpl.rcParams["axes.labelsize"] = 18
+    mpl.rcParams["legend.fontsize"] = 18
+    mpl.rcParams["figure.titlesize"] = 20
+
+
+##### Plotting functions for the dataframe #####
+font_path_ttf = finding_path_to_font("Averta-Regular")
+
+
+######### 1. Distribution of posts over time #########
+def plot_post_distribution_over_time(
+    dataframe: pd.DataFrame, output_path: str, color: str
+) -> None:
```

Comment:
no need for the `None`
```suggestion
):
```